01-basic-java-syntax.md

```diff
@@ -96,47 +125,67 @@ A class variable is also known as a static field. An instance/member variable is
+When you do not add a constructor to a class, the Java compiler adds one for you. The constructor that is added by the Java compiler is called a `default constructor`. The default constructor is public and accepts no arguments, and does nothing. The name of the constructor of a class is the same as the class name.
+
+The default constructor looks like this:
+```
+public Human(){
+
+}
+```
 
 *The general syntax for a method declaration is of the form:*
+
 ```java
 <<modifiers>> <<return type>> <<method name>> (<<parameters list>>) <<throws clause>> {
         // Body of the method goes here
 }
 ```
 
-TIP: Write the methods you wish you had - just pretend they exist. Write the method's name - then hotkey (opt + Return) to generate.
+#### TIP: Error-driven development
+
+Write the stuff you wish you had - just pretend it exist. The IDE (IntelliJ) will then complain at you. For example, you can write the name of a method you wish you had - then hotkey (opt + Return) to generate.
+
+### `this` keyword
+
+The `this` keyword is a reference to the current instance of a class. It can be used only in the context of an instance, e.g. it cannot be used in static methods.
 
-#### `this` keyword is a reference to the current instance of a class. It can be used only in the context of an instance.
 ```java
-public class ThisTest1 {
+public class MyClass {
         int varA = 555;
         int varB = varA;      // Assign value of varA to varB
         int varC = this.varA; // Assign value of varA to varC
 }
 ```
+A lot of the time you can get away with not using the `this.` prefix. It's up to the personal preference/project style guide when you should use it.
+
+### A constructor is a named block of code that is used to initialize an object of a class immediately after the object is created.
 
-#### A constructor is a named block of code that is used to initialize an object of a class immediately after the object is created.
 - The structure of a constructor looks similar to a method.
 - A constructor does not have a return type.
 *The general syntax for a constructor declaration is:*
+
 ```java
 <<Modifiers>> <<Constructor Name>>(<<parameters list>>){
         // Body of constructor goes here
+        // No return keyword
 }
 ```
 
-*Pro tip: Don't ever throw exceptions in your constructor. It's a sign that the constructor is doing more than just setting up it's initial state. (e.g. making database connections).*
+#### TIP: Keep your constructors safe.
+
+You will occasionally have to write code that might not work, based on some external factors (e.g. the computer you're currently using, the input parameters of your method etc.). Avoid this type of code in constructors, as this is bad design. It's a clear sign that the constructor is doing more than just setting up it's initial state. (e.g. making database connections). It's better to only do part of the initialisation in a constructor, and do the more dangerous initialisation in a separate method.
```

Comment:
etc. <- this dot serves a double duty and completes the sentence, so no further dot is necessary!
`... your method etc.) Avoid this...`